### ISSUE TITLE
DM decrypt failed

### ISSUE THREAD

**Issue by believethehype** [holgern/pynostr#16]
I'm getting padding errors when I apply the decrypt function for EncryptedMessages. Do I need a specific version of the dependencies? Is this library still being actively developed?

**Comment by bnonni** [comment]
@believethehype yeah, getting the same error. This is how I resolved it. This code assumes you have subscribed to a relay and set a var called `event` where you are the recipient and someone else is the sender.

```python
your_private_key = private_key.hex()
your_public_key = your_private_key.public_key.hex()

dm = EncryptedDirectMessage(reference_event_id=event.id)
dm.decrypt(private_key.hex(), event.content, public_key.hex())
print(dm.cleartext_content)
```

The package has the naming of the vars in the `EncryptedDirectMessage` mixed up, I think. In a scenario where you are the recipient and someone else is the sender of a DM and you want to instantiate the `EncryptedDirectMessage` class to decrypt the DM, you need to swap the `pubkey` (which should be the senders pubkey) and the `recipient_pubkey` (which should be your pubkey) and it will work. Obviously, this doesn't make sense. I'll be submitting a PR for it.

**Comment by bnonni** [comment]
https://github.com/holgern/pynostr/pull/20
